Extensions/Leaderboards/leaderboardstools.ts

```diff
@@ -45,74 +116,196 @@ namespace gdjs {
       }
       _loaderContainer.appendChild(_loader);
 
-      export const setPlayerScore = function (
+      const getLastScoreSavingData = function ({
+        hasEnded,
+      }: {
+        hasEnded: boolean;
+      }): ScoreSavingData | null {
+        const dateFieldToUse = hasEnded
+          ? 'lastScoreSavingSucceededAt'
+          : 'lastScoreSavingStartedAt';
+        const scoreSavingDataArray = Object.values(
+          _savingDataByLeaderboard
+        ).filter((scoreSavingData) => !!scoreSavingData[dateFieldToUse]);
+        if (scoreSavingDataArray.length === 0) return null;
+
+        let lastScoreSavingData = scoreSavingDataArray[0];
+        scoreSavingDataArray.forEach((scoreSavingData) => {
+          const currentItemDate = scoreSavingData[dateFieldToUse];
+          const lastItemDate = lastScoreSavingData[dateFieldToUse];
```

Comment:
Same here. If you really don't like repeat the condition (which I can understand, this is making the code harder to read), use a closure (see other comment)